src/test/daml/Daml/Finance/Asset/Test/IbericoPigLifeCycle.daml

```diff
@@ -0,0 +1,134 @@
+
+ibericoPigCid <- toInterfaceContractId @Instrument.I <$> submit farmer do createCmd IbericoPig with farmer; id; observers; state; matureDate; cashInstrumentCid
+ibericoPigKey <- Instrument.createReference ibericoPigCid farmer farmer publicObserver
+-- need to read as public as we need holding factory to create new holdings
+transferableIbericoPig: ContractId Transferable.I <- Account.credit [public] ibericoPigKey pigInKg farmerAccount
+
+-- lifecycle ibericoPig for step 1, create pig price observable and pig sold event
+let 
+  settler = bank
+  eventTime = time (matureDate) 12 30 00
+  observations = M.fromList [(dateToDateClockTime (matureDate), 1.0)]
+  lifecycleObserver = [("PigLifecycler", S.singleton $ S.singleton farmer)]
+
+pigPriceOnMatureDateCid <- toInterfaceContractId  <$> submit butcher do 
+  createCmd Observation with provider = butcher; obsKey = show Pig; observations; observers = M.fromList lifecycleObserver
+-- butcherPigPriceOnMatureDateCid <- toInterfaceContractId  <$> submit foodProcessor do createCmd Observation with provider = foodProcessor; obsKey = show Butchered; observations; observers = M.fromList lifecycleObserver
+-- agedHamPriceOnMatureDateCid <- toInterfaceContractId  <$> submit restaurant do createCmd Observation with provider = restaurant; obsKey = show Aged; observations; observers = M.fromList lifecycleObserver
+-- restaurantHamPriceOnMatureDateCid <- toInterfaceContractId  <$> submit restaurant do createCmd Observation with provider = restaurant; obsKey = show Restaurant; observations; observers = M.fromList lifecycleObserver
+
+pigSoldEventCid <- toInterfaceContractId <$> submitMulti [farmer, butcher] [] do 
+  createCmd StateTransitionEvent with currentOwner = farmer; newOwner = butcher; eventTime; currentState = state
+
+-- create clock with mature to declare that current time is mature date
+clockCid <- toInterfaceContractId <$> submit farmer do 
+  createCmd DateClock with u = Unit matureDate; id = show matureDate; provider = farmer; observers = M.empty
+(soldPigCid , [effectCid]) <- Instrument.submitExerciseInterfaceByKeyCmd @Lifecyclable.I [farmer] [public] ibericoPigKey 
+  Lifecyclable.Lifecycle with settler; eventCid = pigSoldEventCid; 
+                                observableCids = [pigPriceOnMatureDateCid]; 
+                                ruleName = show state; clockCid
+
+-- Settlement, create settlement factory, settlement rule and get claim result                                                                                                                                                                   
+factoryCid <- submit farmer do createCmd BatchFactory with provider = farmer; observers = S.empty
+
+settlementRuleCid <- submitMulti [bank, farmer] [] do
+    createCmd Rule
+      with
+        custodian = bank
+        owner = farmer
+        claimers = S.singleton farmer -- Can I make new instrument go into butcher's account? No, not in the current rule implementation
```

Comment:
current SettlementRule does not seem to allow `producedInstrument` to go to other people's account (it will always use the account of `targetInstrument`) 
Should I 
1. use `produced` in the `Effect` instead
2. create a different `SettlementRule` implementation

Reply:
That is right, currently each holding identifies
- a custodian
- an owner

and the life-cycling of an instrument does not change them. This is handled in a separate step (via `Dvp`s).

I feel this represents what happens with the iberico in the real world, e.g.: 
- farmer sells pig to butcher in exchange for cash (this is just a trade, the pig instrument does not change)
- butcher butchers the pig and produces unprocessed ham. This is the lifecycling event (pig changes to ham), owner on the holding does not change (it is still the butcher)
- butcher sells unprocessed meat to food processor (transfer)
- ...